src/Downloader.ts

```diff
@@ -177,16 +183,16 @@ export class Downloader {
         this.browser.close();
     }
 
-    private lastReportedRangeStr() {
+    private lastReportedRangeStr(daysToScrape: number) {
         let reportedRange = '';
-        let detailsSpan = `&detailsSpan=${this.daysToScrape}`;
-        if (this.daysToScrape === 1) {
+        let detailsSpan = `&detailsSpan=${daysToScrape}`;
+        if (daysToScrape === 1) {
             reportedRange = '&lastReportedRange=LAST_24_HRS';
             detailsSpan = '&detailsSpan=7';
-        } else if (this.daysToScrape === 7) {
+        } else if (daysToScrape === 7) {
             reportedRange = '';
         } else {
-            reportedRange = `&lastReportedRange=LAST_${this.daysToScrape}_DAYS`;
+            reportedRange = `&lastReportedRange=LAST_${daysToScrape}_DAYS`;
         }
```

Comment:
Could we use a switch statement to implement this more cleanly?

Reply:
I think the most common practice (not necessarily correct) is to avoid `switch` statements.
Generally if we can convert to a `switch`, we could also convert to a key value store, which is more declarative and nice.

My personal view is that this code isn't too complex, but if it grows more we should definitely do something

Reply:
Thank you :)